### PR TITLE
MNTR-17 Slider shared component added

### PR DESCRIPTION
**Pull request by Gokselagan**
I added slider shared component but ı could not add value in thumb. I tried a few way but it did not work.

### CODE REVIEW

File: src/shared-components/Form/Control/Slider.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+export const CustomSlider = ({
+  defaultValue,
+  color,
+  thumbbgcolor,
+  trackcolor,
+  railcolor,
+}: CustomSliderProps) => {
+  return (
+    <Box sx={{ width: 343 }}>
```
Comment (annasvst) [inline src/shared-components/Form/Control/Slider.tsx:21]: Don't add this wrapper around the slider, let it be as wide as the parent element allow.

File: src/shared-components/Form/Control/Slider.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+interface CustomSliderProps {
+  defaultValue: number;
+  color: string;
+  thumbbgcolor: string;
+  trackcolor: string;
+  railcolor: string;
+}
+
+export const CustomSlider = ({
```
Comment (annasvst) [inline src/shared-components/Form/Control/Slider.tsx:13]: In this component, ask only for one color (base color of the slider. Use that color as color, bg color for thumb and track. For rail, use a `colors.light[20]`

File: src/shared-components/Form/Control/Slider.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  defaultValue,
+  color,
+  thumbbgcolor,
+  trackcolor,
+  railcolor,
+}: CustomSliderProps) => {
```
Comment (annasvst) [inline src/shared-components/Form/Control/Slider.tsx:19]: Remove the other import of the theme, you need to get the theme from context using the `useTheme` hook. Import it form MUI.

```suggestion
}: CustomSliderProps) => {
const theme = useTheme();
```

File: src/shared-components/Form/Control/Slider.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        defaultValue={defaultValue}
+        valueLabelDisplay='auto'
+        aria-label='Default'
+        sx={{
+          color: color,
+          '& .MuiSlider-thumb': {
+            bgcolor: thumbbgcolor,
+            width: 54,
+            height: 30,
+            border: '4px solid #fff',
```
Comment (annasvst) [inline src/shared-components/Form/Control/Slider.tsx:32]:
```suggestion
            border: `${theme.spacing()} solid ${colors.light[100]}`,
```

File: src/shared-components/Form/Control/Slider.tsx (hunk 1 above)
Comment (Gokselagan) [reply src/shared-components/Form/Control/Slider.tsx:21]: Done.

File: src/shared-components/Form/Control/Slider.tsx (hunk 2 above)
Comment (Gokselagan) [reply src/shared-components/Form/Control/Slider.tsx:13]: Fixed

File: src/shared-components/Form/Control/Slider.tsx (hunk 3 above)
Comment (Gokselagan) [reply src/shared-components/Form/Control/Slider.tsx:19]: Fixed

File: src/shared-components/Form/Control/Slider.tsx (hunk 4 above)
Comment (Gokselagan) [reply src/shared-components/Form/Control/Slider.tsx:32]: Fixed

**Review by annasvst — APPROVED** [review]
Approved

**Pull request merged**